src/composables/api/torrents.ts

```diff
@@ -1,19 +1,115 @@
+import { onMounted, ref } from 'vue'
 import { useAxios } from '@vueuse/integrations/useAxios'
 import { axiosInstance } from '@/services/qbit'
 import type { Category } from '@/types/vuetorrent'
+import type { AddTorrentsPayload } from '@/types/vuetorrent/payloads/AddTorrentsPayload'
+import type { QbAddTorrentPayload } from '@/types/qbit/payloads'
 
-export const useCategories = async () => {
-  const { data, error } = await useAxios<Category[]>('/torrents/categories', axiosInstance)
-  if (error) throw new Error(error.value?.message)
-  return { data }
+export const useCategories = () => {
+  const categories = ref<Category[]>()
+
+  async function fetchCategories(): Promise<Category[]> {
+    const { data } = await useAxios<{[key: string]: Category}>('/torrents/categories', axiosInstance)
+    categories.value = Object.values((data.value as {[key: string]: Category}))
+    return categories.value
+  }
+
+  onMounted(() => {
+    fetchCategories()
+  })
+
+  return {
+    fetchCategories,
+    categories,
+  }
 }
 
-export const useAvailableTags = async () => {
-  const { data, error } = await useAxios<string[]>('/torrents/tags', axiosInstance)
+export const useTags = () => {
+
+  const tags = ref<string[]>()
 
-  if (error) throw new Error(error.value?.message)
-  const tags = data.value?.sort((a: string, b: string) =>
-    a.localeCompare(b.toLowerCase(), undefined, { sensitivity: 'base' })
-  )
-  return tags
+  async function fetchTags(): Promise<string[] | undefined> {
+    const { data } = await useAxios<string[]>('/torrents/tags', axiosInstance)
+
+    tags.value = data.value?.sort((a: string, b: string) =>
+      a.localeCompare(b.toLowerCase(), undefined, { sensitivity: 'base' })
+    )
+
+    return data.value
+  }
+
+  onMounted(() => {
+    fetchTags()
+  })
+  
+  return {
+    fetchTags,
+    tags
+  }
 }
+
+export const useTorrents = () => {
+
+  async function addTorrents(payload: AddTorrentsPayload): Promise<any> {
+    
+    let requestBody: FormData | URLSearchParams | null = null;
+    const files: File[] = payload.files
+
+    // Map params
+    const params: QbAddTorrentPayload = {
```

Comment:
This can probably be done a bit more efficient

Reply:
Qbit payload properties do not have a constant format. There's camelCase, snakeCase and so on... I can't see any other way of solving this problem. Do you have a suggestion?